server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -214,6 +226,12 @@ public SearchRequest(StreamInput in) throws IOException {
             finalReduce = true;
         }
         ccsMinimizeRoundtrips = in.readBoolean();
+        if (in.getVersion().onOrAfter(Version.V_7_11_0)) {
+            Boolean readMinCompatVersion = in.readOptionalBoolean();
```

Comment:
We could set the default minVersion to `Version.minimumCompatibilityVersion()` to avoid the null checks ?